tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -165,6 +171,135 @@ struct bt_mesh_cfg_cli_mod_pub pub_params = {
 	.transmit = 0,
 };
 
+extern const struct bt_mesh_comp comp;
+/* For legacy adv, pb-gatt advs are sent with a 1000ms interval. For ext adv, they are sent
+ * with a 100ms interval.
+ */
+static struct bt_mesh_test_gatt gatt_param = {
+#if defined(CONFIG_BT_EXT_ADV)
+	/* (total transmit duration) / (transmit interval) */
+	.transmits = 1500 / 100,
+	.interval = 100,
+#else
+	.transmits = 2000 / 1000,
+	.interval = 1000,
+#endif
+	.service = MESH_SERVICE_PROVISIONING,
+};
+
+static bool gatt_check_rx_count(uint8_t transmit)
+{
+	static int cnt;
+
+	LOG_INF("rx: cnt(%d)", cnt);
+	cnt++;
+
+	if (cnt >= transmit) {
+		cnt = 0;
+		return true;
+	}
+
+	return false;
+}
+
+static void gatt_scan_cb(const bt_addr_le_t *addr, int8_t rssi,
+			  uint8_t adv_type, struct net_buf_simple *buf)
+{
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_IND) {
+		return;
+	}
+
+	/* Ensure that no message is received while Mesh is suspended or disabled. */
+	ASSERT_FALSE_MSG(suspended, "Received adv while Mesh is suspended.");
+
+	parse_mesh_gatt_preamble(buf);
+
+	if (gatt_param.service == MESH_SERVICE_PROVISIONING) {
+		LOG_INF("Parsing pb_gatt adv");
+		parse_mesh_pb_gatt_service(buf);
+	} else {
+		LOG_INF("Parsing proxy adv");
+		parse_mesh_proxy_service(buf);
+	}
+
+	if (gatt_check_rx_count(gatt_param.transmits)) {
+		LOG_INF("rx completed, stopping scan...");
+		k_sem_give(&publish_sem);
+	}
+}
+
+static void gatt_rx_beacons(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type       = BT_HCI_LE_SCAN_PASSIVE,
+		.options    = BT_LE_SCAN_OPT_NONE,
+		.interval   = BT_MESH_ADV_SCAN_UNIT(1000),
+		.window     = BT_MESH_ADV_SCAN_UNIT(1000)
+	};
+	int err;
+
+	err = bt_le_scan_start(&scan_param, gatt_scan_cb);
+	ASSERT_FALSE_MSG(err && err != -EALREADY, "Starting scan failed (err %d)\n", err);
+
+	err = k_sem_take(&publish_sem, K_SECONDS(10));
+	if (err && !suspended) {
+		FAIL("rx timed out");
+	} else if ((err == -EAGAIN) && suspended) {
+		LOG_WRN("rx timed out (tx is suspended)");
+	}
+
+	err = bt_le_scan_stop();
+	ASSERT_FALSE_MSG(err && err != -EALREADY, "Stopping scan failed (err %d)\n", err);
+}
+
+static void gatt_xmit_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+			struct net_buf_simple *buf)
+{
+	uint8_t length;
+
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_NONCONN_IND) {
+		return;
+	}
+
+	length = net_buf_simple_pull_u8(buf);
+	ASSERT_EQUAL(buf->len, length);
+	ASSERT_EQUAL(length, sizeof(uint8_t) + sizeof("0"));
+	ASSERT_EQUAL(BT_DATA_MESH_MESSAGE, net_buf_simple_pull_u8(buf));
+
+	char *data = net_buf_simple_pull_mem(buf, sizeof("0"));
+
+	if (strncmp(data, "0", strlen("0")) == 0) {
+		suspended = false;
+	} else if (strncmp(data, "1", strlen("1")) == 0) {
+		suspended = true;
+	} else {
+		FAIL("Received unexpected data (%s)", data);
+	}
+
+	LOG_INF("Received %s from dut, setting suspended to %s",
+		data, suspended ? "true" : "false");
+	k_sem_give(&publish_sem);
+}
+
+static void gatt_rx_xmit_adv(void)
```

Comment:
```suggestion
static void suspend_state_change_wait(void)
```
or smth similar? `gatt_rx_xmit_adv` is vague.

Reply:
Changed!